### PR TITLE
feat: add flower host

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR adds a host for the flower deployment protected by basic authorization accessible via Caddy.

### CODE REVIEW

File: tutorcelery/patches/caddyfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{% if CELERY_FLOWER and CELERY_FLOWER_PUBLIC %}
+# Superset
```
Comment (MoisesGSalas) [inline tutorcelery/patches/caddyfile]:
```suggestion
# flower
```

File: tutorcelery/patches/k8s-deployments [hunk 2]
```diff
@@ -76,11 +76,38 @@ spec:
...
+kind: ServiceMonitor
+metadata:
+  labels:
+    app.kubernetes.io/name: flower
+    release: prometheus
+  name: flower
+  namespace: chennai-openedx
+spec:
+  endpoints:
+  - interval: 1s
```
Comment (MoisesGSalas) [inline tutorcelery/patches/k8s-deployments]:
```suggestion
  - interval: 30s
```

File: tutorcelery/patches/k8s-deployments [hunk 3]
```diff
@@ -76,11 +76,38 @@ spec:
...
+{% if CELERY_FLOWER_PROMETHEUS -%}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app.kubernetes.io/name: flower
+    release: prometheus
+  name: flower
+  namespace: chennai-openedx
```
Comment (MoisesGSalas) [inline tutorcelery/patches/k8s-deployments]: use K8S_NAMESPACE

File: tutorcelery/plugin.py [hunk 4]
```diff
@@ -26,9 +26,11 @@
         ("CELERY_LMS_EXPLICIT_QUEUES", {}),
         ("CELERY_CMS_EXPLICIT_QUEUES", {}),
         ("CELERY_FLOWER", False),
+        ("CELERY_FLOWER_PUBLIC", False),
         ("CELERY_FLOWER_HOST", "flower.{{LMS_HOST}}"),
         ("CELERY_FLOWER_DOCKER_IMAGE", "docker.io/mher/flower:2.0.1"),
         ("CELERY_MULTIQUEUE_ENABLED", False),
+        ("CELERY_FLOWER_PROMETHEUS", True),
```
Comment (jfavellar90) [inline tutorcelery/plugin.py]: Can we call it CELERY_FLOWER_SERVICE_MONITOR?

File: README.md [hunk 5]
```diff
@@ -69,6 +69,27 @@ times and performance:
...
+- Number of Tasks Prefetched at Worker
+- Tasks Success Ratio
+- Tasks Failure Ratio
+
+Make sure to enable the ServiceMonitor resource to inform Prometheus to scrape metrics from the flower service
+
+```yaml
+CELERY_FLOWER_PROMETHEUS: true
+```
+
```
Comment (jfavellar90) [inline README.md]: I think this is for another PR: can we list all the plugin options and document them together?

File: tutorcelery/patches/k8s-deployments [hunk 6]
```diff
@@ -76,11 +76,38 @@ spec:
...
+spec:
+  endpoints:
+  - interval: 30s
+    port: metrics
+    honorLabels: true
+    path: /metrics
+    schema: http
+  namespaceSelector:
+    matchNames:
+    - chennai-openedx
```
Comment (jfavellar90) [inline tutorcelery/patches/k8s-deployments]: Colonize that Chennai town, please

File: tutorcelery/plugin.py [hunk 7]
```diff
@@ -26,9 +26,11 @@
         ("CELERY_LMS_EXPLICIT_QUEUES", {}),
         ("CELERY_CMS_EXPLICIT_QUEUES", {}),
         ("CELERY_FLOWER", False),
+        ("CELERY_FLOWER_PUBLIC", False),
```
Comment (jfavellar90) [inline tutorcelery/plugin.py]: Maybe CELERY_FLOWER_EXPOSE_SERVICE?

File: tutorcelery/patches/k8s-deployments [hunk 8]
```diff
@@ -76,11 +76,38 @@ spec:
...
             value: {{CELERY_FLOWER_BASIC_AUTH}}
+
+{% if CELERY_FLOWER_PROMETHEUS -%}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    app.kubernetes.io/name: flower
+    release: prometheus
```
Comment (jfavellar90) [inline tutorcelery/patches/k8s-deployments]: Maybe we should add a note saying that the `release: prometheus` label is the default that Prometheus uses to select service monitors. This won't work for a Prometheus with a different ServiceMonitor label selector

File: tutorcelery/patches/caddyfile [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{% if CELERY_FLOWER and CELERY_FLOWER_PUBLIC %}
+# flower
+{{ CELERY_FLOWER_HOST }}{$default_site_port} {
```
Comment (jfavellar90) [inline tutorcelery/patches/caddyfile:3]: Does flower allow to configure the Basic auth?

Comment (Ian2012) [reply tutorcelery/patches/caddyfile:3]: Basic auth is already configured

File: tutorcelery/patches/k8s-services [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+kind: Service
+metadata:
+  name: flower
+  labels:
+    app.kubernetes.io/name: flower
+spec:
+  type: ClusterIP
+  ports:
+    - port: 5555
+      protocol: TCP
```
Comment (MoisesGSalas) [inline tutorcelery/patches/k8s-services:14]: remember to include the name of the port

**Review by MoisesGSalas — APPROVED** [review]

**Review by jfavellar90 — APPROVED** [review]

**Pull request merged**